x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/aggs/InferencePipelineAggregationBuilder.java

```diff
@@ -0,0 +1,236 @@
+            (p, c, n) -> p.namedObject(InferenceConfigUpdate.class, n, c), INFERENCE_CONFIG);
+        PARSER.declareField(InferencePipelineAggregationBuilder::setGapPolicy, p -> {
+            if (p.currentToken() == XContentParser.Token.VALUE_STRING) {
+                return BucketHelpers.GapPolicy.parse(p.text().toLowerCase(Locale.ROOT), p.getTokenLocation());
+            }
+            throw new IllegalArgumentException(
+                "Unsupported token [" + p.currentToken() + "] parsing inference aggregation " + GAP_POLICY.getPreferredName());
+        }, GAP_POLICY, ObjectParser.ValueType.STRING);
+    }
+
+    private final Map<String, String> bucketPathMap;
+    private String modelId;
+    private InferenceConfigUpdate inferenceConfig;
+    private final SetOnce<ModelLoadingService> modelLoadingService;
+    private BucketHelpers.GapPolicy gapPolicy = BucketHelpers.GapPolicy.SKIP;
+
+    public static InferencePipelineAggregationBuilder parse(SetOnce<ModelLoadingService> modelLoadingService,
+                                                            String pipelineAggregatorName,
+                                                            XContentParser parser) {
+        Tuple<SetOnce<ModelLoadingService>, String> context = new Tuple<>(modelLoadingService, pipelineAggregatorName);
+        return PARSER.apply(parser, context);
+    }
+
+    public InferencePipelineAggregationBuilder(String name, SetOnce<ModelLoadingService> modelLoadingService,
+                                               Map<String, String> bucketsPath) {
+        super(name, NAME, new TreeMap<>(bucketsPath).values().toArray(new String[] {}));
+        this.modelLoadingService = modelLoadingService;
+        this.bucketPathMap = bucketsPath;
+    }
+
+    public InferencePipelineAggregationBuilder(StreamInput in, SetOnce<ModelLoadingService> modelLoadingService) throws IOException {
+        super(in, NAME);
+        modelId = in.readString();
+        bucketPathMap = in.readMap(StreamInput::readString, StreamInput::readString);
+        inferenceConfig = in.readOptionalNamedWriteable(InferenceConfigUpdate.class);
+        gapPolicy = BucketHelpers.GapPolicy.readFrom(in);
+        this.modelLoadingService = modelLoadingService;
+    }
+
+    void setModelId(String modelId) {
+        this.modelId = modelId;
+    }
+
+    void setInferenceConfig(InferenceConfigUpdate inferenceConfig) {
+        this.inferenceConfig = inferenceConfig;
+    }
+
+    void setGapPolicy(BucketHelpers.GapPolicy gapPolicy) {
+        this.gapPolicy = gapPolicy;
+    }
+
+    @Override
+    protected void validate(ValidationContext context) {
+        context.validateHasParent(NAME, name);
+        if (modelId == null) {
+            context.addValidationError("Model Id must be set");
+        }
+        if (gapPolicy != BucketHelpers.GapPolicy.SKIP) {
+            context.addValidationError("gap policy [" + gapPolicy + "] in not valid for [" + NAME + "] aggregation");
```

Comment:
Since it only accepts `skip`, you could just remove the option from the parser and hardcode it.

Reply:
GapPolicy is a common setting, it feels a little underhand to silently ignore the setting and rewrite it to `skip`.

Reply:
I think we'd just make it not something the parser recognized at all. Lots of aggs don't have a gap policy.

Reply:
Yep, I meant removing it from the parser, so it'd throw an exception if the user tried to set it.  Most pipeline aggs have a gap_policy but it's not a required feature of pipelines by any means.

Reply:
👍  I assumed pipeline aggs must accept a gap_policy but that doesn't make sense in every case. 

The code was silly anyway I was parsing the field validating it then never reading the setting again. It's gone now.